Tests/NIOTests/NonBlockingFileIOTest.swift

```diff
@@ -915,4 +915,24 @@ class NonBlockingFileIOTest: XCTestCase {
             XCTAssertEqual(numberOfChunks, numberOfCalls.load())
         })
     }
+    
+    func testThrowsErrorOnUnstartedPool() throws {
+        
+        try withTemporaryFile(content: "hello, world") { fileHandle, path in
+            
+            let expectation = XCTestExpectation(description: "Opened file")
+            let threadPool = NIOThreadPool(numberOfThreads: 1)
+            let fileIO = NonBlockingFileIO(threadPool: threadPool)
+            let eventLoopGroup = MultiThreadedEventLoopGroup(numberOfThreads: 1)
+            fileIO.openFile(path: path, eventLoop: eventLoopGroup.next()).whenFailure { (error) in
+                XCTAssertTrue(error is NIOThreadPoolError.ThreadPoolInactive)
+                expectation.fulfill()
+            }
+            
+            XCTAssertNoThrow(try eventLoopGroup.syncShutdownGracefully())
```

Comment:
It would be better to move this to a `defer` block so that it's clear that this shutdown isn't material to the logic of the test itself.